classes/repos/user_repo.php

```diff
@@ -395,4 +397,35 @@ public static function get_mentors_of_user($user)
         return $DB->get_records_list('user', 'id', array_keys($result));
     }
 
+    /**
+     * Returns and array of mentor emails that are assigned to the "mentee" through profile field prepends
+     * 
+     * @param object $user
+     * @return array fake mentor users
+     */
+    public static function get_prepended_mentors_of_user($user)
+    {
+        $users = array();
+
+        profile_load_custom_fields($user);
+
+        foreach($user->profile as $field => $value) {
+            if (! is_string($value)) {
+                return [];
+            }
+            
+            $email = preg_replace('/\s/', '', $value);
+            if (strpos($field, "mentor") > -1) {
```

Comment:
Here it looks like any profile data field with the word "mentor" in it would be assumed to be a mentor's email address. Since these user profile fields are global and used throughout moodle, there is a good possibility that there is a field that matches this assumption and may be representing something entirely different.

Reply:
Would "qm_mentor_" be better?

Reply:
I'm writing up an idea in conversation right now. Gimme just a minute and I'll post it